### PR TITLE
Add EventPipe array types to the profiler APIs

### PR DESCRIPTION
**Pull request by davmason**
Follow up for #36242. Allow profilers to emit array types as well.

### CODE REVIEW

File: src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs [hunk 1]
```diff
@@ -62,21 +64,26 @@ public static int RunTest()
 
                         if (traceEvent.EventName == "AllTypesEvent")
                         {
-                            success = ValidateAllTypesEvent(traceEvent);
+                            success |= ValidateAllTypesEvent(traceEvent);
```
Comment (noahfalk) [inline src/coreclr/tests/src/profiler/eventpipe/eventpipe.cs:67]:
```suggestion
                            success &= ValidateAllTypesEvent(traceEvent);
```

I assume you success should become false if Validate returns false? Same issue a few more times below

File: src/coreclr/tests/src/profiler/native/eventpipeprofiler/eventpipeprofiler.cpp [hunk 2]
```diff
@@ -40,21 +40,21 @@ HRESULT EventPipeProfiler::Initialize(IUnknown* pICorProfilerInfoUnk)
...
-        { COR_PRF_EVENTPIPE_INT32, WCHAR("Int32") },
-        { COR_PRF_EVENTPIPE_UINT32, WCHAR("UInt32") },
-        { COR_PRF_EVENTPIPE_INT64, WCHAR("Int64") },
-        { COR_PRF_EVENTPIPE_UINT64, WCHAR("UInt64") },
-        { COR_PRF_EVENTPIPE_SINGLE, WCHAR("Single") },
-        { COR_PRF_EVENTPIPE_DOUBLE, WCHAR("Double") },
-        { COR_PRF_EVENTPIPE_GUID, WCHAR("Guid") },
-        { COR_PRF_EVENTPIPE_STRING, WCHAR("String") },
-        { COR_PRF_EVENTPIPE_DATETIME, WCHAR("DateTime") }
+        { 0, COR_PRF_EVENTPIPE_BOOLEAN,  WCHAR("Boolean") },
```
Comment (noahfalk) [inline src/coreclr/tests/src/profiler/native/eventpipeprofiler/eventpipeprofiler.cpp:43]:
```suggestion
        { COR_PRF_EVENTPIPE_BOOLEAN, 0, WCHAR("Boolean") },
```
I think the field order is swapped, typecode is first

**Review by noahfalk — APPROVED** [review]
LGTM (with a few commented suspicious things in the tests)

**Comment by davmason** [comment]
> 
> 
> LGTM (with a few commented suspicious things in the tests)

In case it wasn't obvious I was testing outside of the repo so I could use the latest TraceEvent and validate that it correctly parsed my array type metadata generation. Then I reworked a bunch of things and forgot to update the test in the tree.

**Pull request merged**